pkg/controllers/nodeclaim/lifecycle/liveness.go

```diff
@@ -103,3 +130,16 @@ func (l *Liveness) updateNodePoolRegistrationHealth(ctx context.Context, nodeCla
 	}
 	return nil
 }
+
+func (l *Liveness) deleteNodeClaimForTimeout(ctx context.Context, timeout NodeClaimTimeout, nodeClaim *v1.NodeClaim) error {
+	if err := l.kubeClient.Delete(ctx, nodeClaim); err != nil {
+		return client.IgnoreNotFound(err)
+	}
+	log.FromContext(ctx).V(1).WithValues("timeout", timeout.duration).Info(fmt.Sprintf("terminating due to %s timeout", timeout.reason))
```

Comment:
Since you are formatting the log line here, that sort-of implies that this could be a structured value -- can you put this value in the `WithValues()` as a `reason`?